src/Compilers/CSharp/Portable/Binder/ContextualAttributeBinder.cs

```diff
@@ -64,5 +66,14 @@ internal Symbol AttributeTarget
                 return _attributeTarget;
             }
         }
+
+        protected override CSDiagnosticInfo SingleTypeNameNotFound(string whereText, DiagnosticBag diagnostics, Location location, AssemblySymbol forwardedToAssembly)
+        {
+            if (!whereText.EndsWith("Attribute"))
```

Comment:
@jskeet Seems the legacy compiler just blindly appends `Attribute` to everything. Will Update.